src/common/blkdev.cc

```diff
@@ -429,6 +429,14 @@ std::string _decode_model_enc(const std::string& in)
     v.erase(found + 1);
   }
   std::replace(v.begin(), v.end(), ' ', '_');
+
+  // remove "__", which seems to come up on by ubuntu box for some reason.
+  while (true) {
+    auto p = v.find("__");
+    if (p == std::string::npos) break;
+    v.replace(p, 2, "_");
+  }
```

Comment:
i'd use `boost::replace_all()`. see https://www.boost.org/doc/libs/1_76_0/doc/html/boost/algorithm/replace_all.html

actually, we've been using it in Ceph already. 
```console
% git grep -w replace_all
src/auth/KeyRing.cc:      boost::replace_all(caps, "\"", "\\\"");
src/common/ceph_context.cc:    boost::replace_all(section, " ", "_");
src/libradosstriper/RadosStriperImpl.cc:    boost::replace_all(format, "%", "%%");
src/libradosstriper/RadosStriperImpl.cc:    boost::replace_all(format, "%", "%%");
src/rgw/rgw_auth_s3.cc:    boost::replace_all(copy_params, "+", "%20");
src/rgw/rgw_auth_s3.cc:      boost::replace_all(_val, "+", " ");
src/rgw/rgw_auth_s3.h:    boost::replace_all(canonical_uri, "+", "%20");
src/rgw/rgw_auth_s3.h:    boost::replace_all(canonical_uri, "+", "%20");
src/rgw/rgw_rest_s3.cc:    boost::replace_all(key, "-", "_");
src/rgw/rgw_rest_s3.cc:    boost::replace_all(m_s3select_query, GT, ">");
src/rgw/rgw_rest_s3.cc:    boost::replace_all(m_s3select_query, LT, "<");
src/tools/rbd/action/Perf.cc:  boost::replace_all(s, "_", " ");
src/tools/rbd/action/Perf.cc:  boost::replace_all(s, "-", "_");
```

Reply:
Tried that, but replace_all won't turn 3 _'s into 1.

Reply:
i see. what about `std::regex_replace()` ?